SimplenoteUITests/Generic.swift

```
@@ -47,6 +47,23 @@ func toggleSwitchIfNeeded(_ switchElement: XCUIElement, _ value: String) {
     switchElement.tap()
 }
 
+func dismissVerifyEmailIfNeeded(using app: XCUIApplication) {
+    guard app.staticTexts["Review Your Account"].waitForExistence(timeout: 15) else { return }
+    app.buttons["icon cross"].tap()
+}
+
+func populateNoteList(title: String, numberToCreate: Int) {
+    for i in 1...numberToCreate {
+        let noteTitle = title + String(i)
+        let noteContent = "\n\nSome content here for note " + String(i)
+
+        NoteList.addNoteTap()
+        NoteEditorAssert.editorShown()
```

Philosophical topic: what do you think of having assertions in non-asserting methods?

Currently, in Simplenote iOS, we have assertions really exposed - they are either in the test body, or in the methods that contain `Assert` is their name / class. This way, we know that we have a set of non-asserting methods, that do stuff, but do not care how correctly it goes. And the other set of asserting methods, that care only about how correctly something went.

Also, in that particular case, this method is expected to execute a loop of actions, and repeat the assertion `i` times - which means spending 5 * `i` seconds to assert something, that is already asserted in other tests of `Note Editor`.

^ This is only my opinion, and I can also see value is having as many assertions as possible. The main concern here is time: how many tests can we run on CI under 5 minutes (this is just an approximately acceptable time our developers are okay with, if we talk about running a subset of tests on every commit to PR - which happens now).

---

Yup makes sense, I've used it sometimes on non-assert methods to make test fail faster, but usually prefer to have it separated as well. In this case, I do agree that it's not really needed/add much value being there. Will remove it.